src/main/ruby/core/truffle/interop.rb

```diff
@@ -146,12 +146,31 @@ def self.to_java_array(array)
       to_java_array(Truffle::Type.coerce_to(array, ::Array, :to_a))
     end
     
+    def self.special_form(receiver, name, *args)
+      case name.to_sym
+      when :delete
+        Truffle::Interop.remove(receiver, *args)
+      when :size
+        Truffle::Interop.size(receiver)
+      when :keys
+        Truffle::Interop.keys(receiver)
+      else
+        raise
+      end
```

Comment:
It seems the node could call directly the corresponding method on `Truffle::Interop` to avoid overhead.

Reply:
Yes - I was thinking they'd be more complex when I started this. Won't change now.